### PR TITLE
doc: Add Pacific release notes

### PR DESCRIPTION
**Pull request by djgalloway**
This is my first stab at Release Notes for a new major release so I'll need some hand holding.  I just copied `octopus.rst` and modified the stuff I knew about.. mainly just https://github.com/ceph/ceph/blob/octopus/PendingReleaseNotes.

1. Do we need to split **Notable Changes** out by component?  If so, I need each lead to provide bullet points for their component.
2. In `octopus.rst`, the **Notable Changes** are under "[Upgrade compatibility notes](https://github.com/ceph/ceph/blob/octopus/doc/releases/octopus.rst#upgrade-compatibility-notes)."  That seems wrong so I named it "Notable Changes" in this file.
3. I guessed on what versions to replace under **Upgrading from pre-...** See https://github.com/ceph/ceph/blob/octopus/doc/releases/octopus.rst#upgrading-from-pre-mimic-releases-like-luminous for Octopus'

Basically I need help filling everything else out besides **Changelog** and **Notable Changes**

### CODE REVIEW

**Review by rishabh-d-dave — COMMENTED** [review]
@batrick Nothing besides nits.

**Comment by petrutlucian94** [comment]
Do you think it would make sense to have a line on Windows support? For example: "Windows hosts can now natively consume Ceph resources such as rados, rbd or cephfs".

File: doc/releases/pacific.rst [hunk 1]
```diff
@@ -51,7 +51,40 @@ RGW object storage
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* cephfs-mirror...
+* The CephFS adds modifies on-RADOS metadata such that the new format is no
+  longer backwards compatible. It is not possible to downgrade a file system from
+  Pacific (or later) to an older release.
+
+* Multiple file systems in a single Ceph cluster is now stable. New Ceph clusters
+  enable support for multiple file systems by default. Existing clusters
+  must still set the "enable_multiple" flag on the fs. Please see the CephFS
```
[the remark this answers is not in the document]
Comment (batrick) [reply doc/releases/pacific.rst]: Thanks, fixed.

File: doc/releases/pacific.rst [hunk 2]
```diff
@@ -51,7 +51,40 @@ RGW object storage
 CephFS distributed file system
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* cephfs-mirror...
+* The CephFS adds modifies on-RADOS metadata such that the new format is no
```
[the remark this answers is not in the document]
Comment (batrick) [reply doc/releases/pacific.rst]: Thanks, fixed.

File: doc/releases/pacific.rst [hunk 3]
```diff
@@ -51,7 +51,40 @@ RGW object storage
...
+* Multiple active MDS file system scrub is now stable. It is no longer necessary
+  to set ``max_mds`` to 1 and wait for non-zero ranks to stop. Scrub commands
+  can only be sent to rank 0: ``ceph tell mds.<fs_name>:0 scrub start /path ...``.
+
+* Ephemeral pinning -- policy based subtree pinning -- is considered stable.
+  ``mds_export_ephemeral_random`` and ``mds_export_ephemeral_distributed`` now
+  default to true.
+
+* A new cephfs-mirror daemon is available for mirror CephFS file systems to
+  remote Ceph cluster. See the CephFS documentation for details.
```
[the remark this answers is not in the document]
Comment (batrick) [reply doc/releases/pacific.rst]: Thanks, fixed.

**Comment by batrick** [comment]
> Do you think it would make sense to have a line on Windows support? For example: "Windows hosts can now natively consume Ceph resources such as rados, rbd or cephfs".

I added a note for CephFS.

File: doc/releases/pacific.rst [hunk 4]
```diff
@@ -51,7 +51,40 @@ RGW object storage
...
+* Multiple file systems in a single Ceph cluster is now stable. New Ceph clusters
+  enable support for multiple file systems by default. Existing clusters
+  must still set the "enable_multiple" flag on the fs. Please see the CephFS
+  documentation for more information.
+
+* A new ``mds_autoscaler`` plugin is available for automatically deploying
+  MDS daemons in response to changes to the ``max_mds`` configuration. Expect
+  further enhancements in the future to simplify MDS scaling.
+
+* cephfs-top is a new utility for looking at performance metrics from CephFS
```
Comment (rishabh-d-dave) [inline doc/releases/pacific.rst]: Missed out this one last time -
```suggestion
* ``cephfs-top`` is a new utility for looking at performance metrics from CephFS
```

**Review by rishabh-d-dave — COMMENTED** [review]
Looks perfect otherwise.

File: doc/releases/pacific.rst (hunk 4 above)
Comment (rishabh-d-dave) [reply doc/releases/pacific.rst]: @batrick

Comment (batrick) [reply doc/releases/pacific.rst]: done, thanks

**Comment by ajarr** [comment]
> Looks perfect otherwise.

Agree. Looks good, Patrick!

**Comment by idryomov** [comment]
Added `CEPH_X2` note.

@batrick Your "qa: fix formatting of cephfs-top" was breaking the SOB check.  I took the liberty and squashed it into "doc: include more CephFS updates".

**Comment by cbodley** [comment]
i see lots of really old changes in pacific.rst that link to PRs dating back to 2014

File: doc/releases/pacific.rst [hunk 5]
```diff
@@ -133,6 +133,20 @@ RBD block storage
 RGW object storage
 ~~~~~~~~~~~~~~~~~~
 
+* Initial support for S3 Select. See :ref:`s3-select-feature-table` for supported queries.
+
+* Bucket notification topics can be configured as ``persistent``, where events
+  are recorded in rados for reliable delivery.
+
+* Bucket notifications can be delivered to SSL-enabled AMQP endpoints.
+
```
Comment (yuvalif) [inline doc/releases/pacific.rst]: we now support AWS compliant API to get topic attributes: GetTopicAttributes

**Comment by neha-ojha** [comment]
https://ceph--40265.org.readthedocs.build/en/40265/releases/pacific/

File: doc/releases/general.rst [hunk 6]
```diff
@@ -121,6 +121,9 @@ Release timeline
 
 .. ceph_timeline:: releases.yml development octopus nautilus mimic luminous kraken jewel infernalis hammer giant firefly
 
+.. _Pacific: ../pacific
+.. _16.1.0: ../octopus#v16-1-0-pacific
```
Comment (neha-ojha) [inline doc/releases/general.rst:125]:
```suggestion
.. _16.2.0: ../pacific#v16-2-0-pacific
```

File: doc/releases/pacific.rst (hunk 5 above)
Comment (cbodley) [reply doc/releases/pacific.rst:142]: @yuvalif your text from PendingReleaseNotes is already under the `Notable Changes` section, so i didn't repeat it under rgw:
```
* An AWS-compliant API: "GetTopicAttributes" was added to replace the existing "GetTopic" API. The new API
  should be used to fetch information about topics used for bucket notifications.
```

**Review by liewegas — APPROVED** [review]

**Review by neha-ojha — APPROVED** [review]

**Review by idryomov — APPROVED** [review]

**Review by jdurgin — APPROVED** [review]

**Comment by djgalloway** [comment]
@sebastian-philipp are you ok with this as-is?

File: doc/releases/pacific.rst (hunk 5 above)
Comment (yuvalif) [reply doc/releases/pacific.rst:142]: sorry, i missed that. thanks!

**Comment by jc305697** [comment]
Hi, 
I just want you to let you know that there is a little typo in  the following line: "The legal vaues for osd_scrub_begin_week_day and osd_scrub_end_week_day are 0 - 6." 
Correction:
 The legal values for osd_scrub_begin_week_day and osd_scrub_end_week_day are 0 - 6. 

I don't know how to make the change but thought I should let you know.

Thanks for everything you do :) .

**Comment by djgalloway** [comment]
> Hi,
> I just want you to let you know that there is a little typo in the following line: "The legal vaues for osd_scrub_begin_week_day and osd_scrub_end_week_day are 0 - 6."
> Correction:
> The legal values for osd_scrub_begin_week_day and osd_scrub_end_week_day are 0 - 6.
> 
> I don't know how to make the change but thought I should let you know.
> 
> Thanks for everything you do :) .

Thank you!  Fixed.
